tests/integration-service/namespace-backed-environments.go

```diff
@@ -121,6 +121,9 @@ var _ = framework.IntegrationServiceSuiteDescribe("Namespace-backed Environment
 		It("checks for deploymentTargetClaim after Ephemeral env has been created", func() {
 			Eventually(func() error {
 				dtcl, err = f.AsKubeDeveloper.GitOpsController.GetDeploymentTargetClaimsList(testNamespace)
+				if err != nil {
+					return fmt.Errorf("failed to find deploymentTargetClaim: %w", err)
+				}
 				Expect(err).ToNot(HaveOccurred())
```

Comment:
@hongweiliu17 I think that the absence of this check here was intentional.
Because we didn't want to retry in case we have trouble fetching the DTC list from the cluster, because that error is unexpected. This will just increase the test duration (due to retries) in case there's something wrong with the cluster, which might be misleading as it should fail the test on the first go instead.

Ideally, we should only retry expected (non-infra-related) errors which we know will get fixed if we wait a little more. For example: waiting for the DTC's phase is not yet equal to "Bound", or waiting for a resource's status condition to turn from "Waiting" to "Provisioned", etc. But I'm fine with having this change here :+1:

Reply:
Good point, we should not retry for the unrecoverable error.

Reply:
Yes, that's correct.
However, if you want to add some Assertions within the Eventually blocks that _does_ perform retries, you can pass the Gomega interface and use `g.Expect()` instead, these assertions will retry on failure.

More on this here (at the end of [this section](https://onsi.github.io/ginkgo/#testing-external-systems)):
![image](https://github.com/redhat-appstudio/e2e-tests/assets/43273420/80ce1cab-6725-422f-963a-b54d12626a90)

Reply:
Thanks! Let's stop the retry here to save some time.

Reply:
https://github.com/redhat-appstudio/e2e-tests/pull/985 is created.

Reply:
Thanks!